packages/firebase_core/firebase_core_platform_interface/test/method_channel_firebase_core_test.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:firebase_core_platform_interface/firebase_core_platform_interface.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  group('$MethodChannelFirebaseCore', () {
+    final MethodChannelFirebaseCore channelPlatform =
+        MethodChannelFirebaseCore();
+    final List<MethodCall> log = <MethodCall>[];
+    const FirebaseOptions testOptions = FirebaseOptions(
+      apiKey: 'testAPIKey',
+      bundleID: 'testBundleID',
+      clientID: 'testClientID',
+      trackingID: 'testTrackingID',
+      gcmSenderID: 'testGCMSenderID',
+      projectID: 'testProjectID',
+      androidClientID: 'testAndroidClientID',
+      googleAppID: 'testGoogleAppID',
+      databaseURL: 'testDatabaseURL',
+      deepLinkURLScheme: 'testDeepLinkURLScheme',
+      storageBucket: 'testStorageBucket',
+    );
+    final FirebaseAppData testApp = FirebaseAppData('testApp', testOptions);
+
+    setUp(() async {
+      MethodChannelFirebaseCore.channel
+          .setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        switch (methodCall.method) {
```

Comment:
Are this tests new, or taken from the current implementation of the plugin?

Reply:
They are basically copies of the tests in `firebase_core`